cmd/ocsp-updater/main.go

```diff
@@ -220,40 +328,41 @@ func main() {
 
 		cac, closeChan := setupClients(c, stats)
 
+		updater, err := newUpdater(
+			auditlogger,
+			stats,
+			clock.Default(),
+			dbMap,
+			cac,
+			// Necessary evil for now
+			c.OCSPUpdater,
+		)
+		cmd.FailOnError(err, "Failed to create updater")
+
 		go func() {
-			// Abort if we disconnect from AMQP
-			for {
-				for err := range closeChan {
-					auditlogger.Warning(fmt.Sprintf(" [!] AMQP Channel closed, aborting early: [%s]", err))
-					panic(err)
-				}
-			}
+			err := <-closeChan
+			// TODO(): When the channel falls over so do we for now, if the AMQP channel
+			// has already closed there is no real cleanup we can do. This is due to
+			// really needing to change the underlying AMQP Server/Client reconnection
+			// logic.
+			auditlogger.AuditErr(fmt.Errorf(" [!] AMQP Channel closed, exiting: [%s]", err))
+			os.Exit(1)
 		}()
 
-		updater := &OCSPUpdater{
-			cac:   cac,
-			dbMap: dbMap,
-			stats: stats,
-			log:   auditlogger,
-		}
-
-		// Calculate the cut-off timestamp
-		if c.OCSPUpdater.MinTimeToExpiry == "" {
-			panic("Config must specify a MinTimeToExpiry period.")
-		}
-		dur, err := time.ParseDuration(c.OCSPUpdater.MinTimeToExpiry)
-		cmd.FailOnError(err, "Could not parse MinTimeToExpiry from config.")
+		go updater.newCertificatesLoop.loop()
+		go updater.oldOCSPResponsesLoop.loop()
 
-		oldestLastUpdatedTime := time.Now().Add(-dur)
-		auditlogger.Info(fmt.Sprintf("Searching for OCSP responses older than %s", oldestLastUpdatedTime))
+		// Catch INT/TERM/HUP signals and exit after work is done
+		sigChan := make(chan os.Signal, 1)
```

Comment:
This block of code seems to not be needed anymore if we're not worried about where this app gets terminated cleanly.